### ISSUE TITLE
Add Option to Disable `prefer-destructuring` When the Variable is Already Declared

### ISSUE THREAD

**Issue by nahtnam** [eslint/eslint#10782]
**What rule do you want to change?**
I would like to propose a change to `prefer-destructuring`.

**Does this change cause the rule to produce more or fewer warnings?**
Fewer warnings if configured.

**How will the change be implemented? (New option, new default behavior, etc.)?**
A new option to `prefer-destructuring` along the lines of `ignore-declared-variables`.

**Please provide some example code that this change will affect:**

```js
let status;
// ... some code here
status = response.status; // eslint complains here
```

To fix the above issue, we'd have to do this:
```js
let status;
// ... some code here
({ status } = response);
```
I feel like the above code is much more confusing than the first piece of code.

**What does the rule currently do for this code?**
```
error  Use object destructuring  prefer-destructuring
```
**What will the rule do after it's changed?**
If configured, it should not complain about needing to be destructred if and only if the variable is declared previously.

**Comment by bendtherules** [comment]
Correct me if I didn't understand the issue, but shouldn't this suffice :
```js
let status;
/* eslint prefer-destructuring: ["error", {VariableDeclarator: {object: true}}] */
status = response.status;
```
Basically, set the option `AssignmentExpression` false for both array and objects.

**Comment by nahtnam** [comment]
Yep, that did the trick, thank you!
